EIPS/eip-2935.md

```diff
@@ -58,44 +59,46 @@ For resolving the `BLOCKHASH` opcode this fork onwards (`block.timestamp >= FORK
 ```
 
-Note that the above logic allows access deeper than `MIN_HISTORY_SERVE_WINDOW` if it exists all the way upto `MAX_HISTORY_SERVE_WINDOW`.
-
-Edge cases:
+Some activation scenarios:
 
  * For the fork to be activated at genesis, no history is written to the genesis state, and at the start of block `1`, genesis hash will be written as a normal operation to slot `0`.
  * for activation at block `1`, only genesis hash will be written at slot `0` as there is no additional history that needs to be persisted.
  * for activation at block `32`, block `31`'s hash will be written to slot `31` and additonal history for `0..30`'s hashes will be persisted, so all in all `0..31`'s hashes.
- * for activation at block `1000`, block `744-999`'s hashes will be presisted in the slot and `BLOCKHASH` for `733` or less would resolve to `0` as only `MIN_HISTORY_SERVE_WINDOW` is persisted.
+ * for activation at block `1000`, block `744-999`'s hashes will be presisted in the slot and `BLOCKHASH` for `733` or less would resolve to `0` as only `HISTORY_SERVE_WINDOW` can be served.
 
-### [EIP-158](./eip-158.md) exception
+### [EIP-158](./eip-158.md) handling
 
-This address is currently exempt from [EIP-158](./eip-158.md) cleanup in Kaustinen Verkle Testnet but there are two ways this could be addressed before this EIP is adopted by ACD:
+This address is currently exempt from [EIP-158](./eip-158.md) cleanup in Kaustinen Verkle Testnet but we plan to address this in the following way:
 
-* Update the nonce to 1 in the fork block, or
-* Deploy a contract à la [EIP-4788](./eip-4788.md) with `BLOCKHASH` opcode delegating call to this contract with appropriate args.
-
-While the second option looks more elegant, it has a higher complexity as well as gas consumption considerations.
+* Deploy a contract à la [EIP-4788](./eip-4788.md) which just supports `get` method to resolve the BLOCKHASH as per the logic defined above (and use the generated address as the BLOCKHASH contract address).
+* While the clients are expected to directly read from state (or maintain and server from memory) to resolve BLOCKHASH opcode, this contract's `get` could be invoked by transaction (via contract or otherwise) leading to a normal contract execution (and gas consumption) as per the semantics of the contract call.
 
 ## Rationale
 
 Very similar ideas were proposed before in [EIP-210](./eip-210.md) et al. This EIP is a simplification, removing two sources of needless complexity:
 
 1. Having a tree-like structure with multiple layers as opposed to a single list
 2. Writing the EIP in EVM code
+3. Serial unbounded storage of hashes for a deep access to the history
+
+However after weighing pros and cons, we decided to go with just a limited ring buffer to only serve the requiste `HISTORY_SERVE_WINDOW` as [EIP 4788](./eip04788) and beacon state accumulators allow (albit a bit more complex) proof against any ancestor since merge.
```

Comment:
```suggestion
However after weighing pros and cons, we decided to go with just a limited ring buffer to only serve the requiste `HISTORY_SERVE_WINDOW` as [EIP-4788](./eip04788) and beacon state accumulators allow (albit a bit more complex) proof against any ancestor since merge.
```